### PR TITLE
DOC: Fix flake8 issues in categorical.rst

### LINKED ISSUE #23793: DOC: Fix format of categorical.rst
Our documentation should follow the PEP-8 standard, but for historical reasons there are many formats that don't. We should fix them now.

This issue is to fix the format of `doc/source/categorical.rst`. To find the errors, first is required to install a newer version of `flake8-rst`:
```
conda install -c conda-forge flake8-rst=0.5.0
```

Then, all the errors can be obtained with:
```
$ flake8-rst doc/source/categorical.rst
doc/source/categorical.rst:63:23: E231 missing whitespace after ','
doc/source/categorical.rst:63:27: E231 missing whitespace after ','
doc/source/categorical.rst:63:31: E231 missing whitespace after ','
doc/source/categorical.rst:70:27: E231 missing whitespace after ':'
doc/source/categorical.rst:70:32: E231 missing whitespace after ','
doc/source/categorical.rst:70:36: E231 missing whitespace after ','
doc/source/categorical.rst:70:40: E231 missing whitespace after ','
doc/source/categorical.rst:89:34: E231 missing whitespace after ','
doc/source/categorical.rst:89:38: E231 missing whitespace after ','
doc/source/categorical.rst:89:42: E231 missing whitespace after ','
doc/source/categorical.rst:89:64: E231 missing whitespace after ','
doc/source/categorical.rst:89:68: E231 missing whitespace after ','
doc/source/categorical.rst:93:27: E231 missing whitespace after ':'
doc/source/categorical.rst:93:32: E231 missing whitespace after ','
doc/source/categorical.rst:93:36: E231 missing whitespace after ','
doc/source/categorical.rst:93:40: E231 missing whitespace after ','
doc/source/categorical.rst:157:5: E402 module level import not at top of file
doc/source/categorical.rst:189:35: E231 missing whitespace after ','
doc/source/categorical.rst:189:49: E231 missing whitespace after ','
doc/source/categorical.rst:190:84: E501 line too long (80 > 79 characters)
doc/source/categorical.rst:201:23: E231 missing whitespace after ','
doc/source/categorical.rst:201:27: E231 missing whitespace after ','
doc/source/categorical.rst:201:31: E231 missing whitespace after ','
doc/source/categorical.rst:238:4: E402 module level import not at top of file
doc/source/categorical.rst:271:32: E241 multiple spaces after ','
doc/source/categorical.rst:296:29: E231 missing whitespace after ':'
doc/source/categorical.rst:296:38: E231 missing whitespace after ':'
doc/source/categorical.rst:312:23: E231 missing whitespace after ','
doc/source/categorical.rst:312:27: E231 missing whitespace after ','
doc/source/categorical.rst:312:31: E231 missing whitespace after ','
doc/source/categorical.rst:320:38: E231 missing whitespace after ','
doc/source/categorical.rst:320:42: E231 missing whitespace after ','
doc/source/categorical.rst:320:46: E231 missing whitespace after ','
doc/source/categorical.rst:320:68: E231 missing whitespace after ','
doc/source/categorical.rst:320:72: E231 missing whitespace after ','
doc/source/categorical.rst:357:23: E231 missing whitespace after ','
doc/source/categorical.rst:357:27: E231 missing whitespace after ','
doc/source/categorical.rst:357:31: E231 missing whitespace after ','
doc/source/categorical.rst:361:35: E231 missing whitespace after ','
doc/source/categorical.rst:361:37: E231 missing whitespace after ','
doc/source/categorical.rst:381:30: E231 missing whitespace after ','
doc/source/categorical.rst:381:32: E231 missing whitespace after ','
doc/source/categorical.rst:390:30: E231 missing whitespace after ','
doc/source/categorical.rst:390:32: E231 missing whitespace after ','
doc/source/categorical.rst:425:38: E231 missing whitespace after ','
doc/source/categorical.rst:425:42: E231 missing whitespace after ','
doc/source/categorical.rst:425:64: E231 missing whitespace after ','
doc/source/categorical.rst:425:68: E231 missing whitespace after ','
doc/source/categorical.rst:425:72: E231 missing whitespace after ','
doc/source/categorical.rst:439:25: E231 missing whitespace after ','
doc/source/categorical.rst:439:31: E231 missing whitespace after ','
doc/source/categorical.rst:441:36: E231 missing whitespace after ','
doc/source/categorical.rst:441:42: E231 missing whitespace after ','
doc/source/categorical.rst:441:50: E231 missing whitespace after ','
doc/source/categorical.rst:459:38: E231 missing whitespace after ','
doc/source/categorical.rst:459:42: E231 missing whitespace after ','
doc/source/categorical.rst:459:46: E231 missing whitespace after ','
doc/source/categorical.rst:461:23: E231 missing whitespace after ','
doc/source/categorical.rst:461:27: E231 missing whitespace after ','
doc/source/categorical.rst:461:31: E231 missing whitespace after ','
doc/source/categorical.rst:481:21: E231 missing whitespace after ','
doc/source/categorical.rst:481:23: E231 missing whitespace after ','
doc/source/categorical.rst:481:25: E231 missing whitespace after ','
doc/source/categorical.rst:482:32: E231 missing whitespace after ','
doc/source/categorical.rst:482:34: E231 missing whitespace after ','
doc/source/categorical.rst:499:21: E231 missing whitespace after ','
doc/source/categorical.rst:499:23: E231 missing whitespace after ','
doc/source/categorical.rst:499:25: E231 missing whitespace after ','
doc/source/categorical.rst:500:36: E231 missing whitespace after ','
doc/source/categorical.rst:500:38: E231 missing whitespace after ','
doc/source/categorical.rst:529:27: E203 whitespace before ':'
doc/source/categorical.rst:529:78: E231 missing whitespace after ','
doc/source/categorical.rst:529:82: E231 missing whitespace after ','
doc/source/categorical.rst:529:83: E501 line too long (99 > 79 characters)
doc/source/categorical.rst:530:27: E203 whitespace before ':'
doc/source/categorical.rst:530:32: E231 missing whitespace after ','
doc/source/categorical.rst:530:34: E231 missing whitespace after ','
doc/source/categorical.rst:530:36: E231 missing whitespace after ','
doc/source/categorical.rst:530:38: E231 missing whitespace after ','
doc/source/categorical.rst:530:40: E231 missing whitespace after ','
doc/source/categorical.rst:530:42: E231 missing whitespace after ','
doc/source/categorical.rst:530:44: E231 missing whitespace after ','
doc/source/categorical.rst:530:47: E202 whitespace before '}'
doc/source/categorical.rst:537:51: E231 missing whitespace after ','
doc/source/categorical.rst:537:55: E231 missing whitespace after ','
doc/source/categorical.rst:538:27: E231 missing whitespace after ','
doc/source/categorical.rst:563:23: E231 missing whitespace after ','
doc/source/categorical.rst:563:25: E231 missing whitespace after ','
doc/source/categorical.rst:566:28: E231 missing whitespace after ','
doc/source/categorical.rst:566:30: E231 missing whitespace after ','
doc/source/categorical.rst:569:29: E231 missing whitespace after ','
doc/source/categorical.rst:569:31: E231 missing whitespace after ','
doc/source/categorical.rst:589:23: E231 missing whitespace after ','
doc/source/categorical.rst:589:25: E231 missing whitespace after ','
doc/source/categorical.rst:599:10: E111 indentation is not a multiple of four
doc/source/categorical.rst:607:23: E231 missing whitespace after ','
doc/source/categorical.rst:607:25: E231 missing whitespace after ','
doc/source/categorical.rst:612:10: E111 indentation is not a multiple of four
doc/source/categorical.rst:635:38: E231 missing whitespace after ','
doc/source/categorical.rst:635:42: E231 missing whitespace after ','
doc/source/categorical.rst:635:46: E231 missing whitespace after ','
doc/source/categorical.rst:635:68: E231 missing whitespace after ','
doc/source/categorical.rst:635:72: E231 missing whitespace after ','
doc/source/categorical.rst:635:76: E231 missing whitespace after ','
doc/source/categorical.rst:642:31: E231 missing whitespace after ','
doc/source/categorical.rst:642:35: E231 missing whitespace after ','
doc/source/categorical.rst:642:39: E231 missing whitespace after ','
doc/source/categorical.rst:642:43: E231 missing whitespace after ','
doc/source/categorical.rst:642:47: E231 missing whitespace after ','
doc/source/categorical.rst:642:51: E231 missing whitespace after ','
doc/source/categorical.rst:642:73: E231 missing whitespace after ','
doc/source/categorical.rst:642:77: E231 missing whitespace after ','
doc/source/categorical.rst:642:81: E231 missing whitespace after ','
doc/source/categorical.rst:642:84: E501 line too long (82 > 79 characters)
doc/source/categorical.rst:643:30: E231 missing whitespace after ':'
doc/source/categorical.rst:643:35: E231 missing whitespace after ','
doc/source/categorical.rst:643:44: E231 missing whitespace after ':'
doc/source/categorical.rst:643:47: E231 missing whitespace after ','
doc/source/categorical.rst:643:49: E231 missing whitespace after ','
doc/source/categorical.rst:643:51: E231 missing whitespace after ','
doc/source/categorical.rst:643:53: E231 missing whitespace after ','
doc/source/categorical.rst:643:55: E231 missing whitespace after ','
doc/source/categorical.rst:643:57: E231 missing whitespace after ','
doc/source/categorical.rst:646:32: E231 missing whitespace after ','
doc/source/categorical.rst:646:36: E231 missing whitespace after ','
doc/source/categorical.rst:646:40: E231 missing whitespace after ','
doc/source/categorical.rst:646:62: E231 missing whitespace after ','
doc/source/categorical.rst:646:66: E231 missing whitespace after ','
doc/source/categorical.rst:647:31: E231 missing whitespace after ':'
doc/source/categorical.rst:647:37: E231 missing whitespace after ','
doc/source/categorical.rst:647:41: E231 missing whitespace after ':'
doc/source/categorical.rst:647:46: E231 missing whitespace after ','
doc/source/categorical.rst:647:50: E231 missing whitespace after ','
doc/source/categorical.rst:647:54: E231 missing whitespace after ','
doc/source/categorical.rst:647:69: E231 missing whitespace after ':'
doc/source/categorical.rst:647:72: E231 missing whitespace after ','
doc/source/categorical.rst:647:74: E231 missing whitespace after ','
doc/source/categorical.rst:647:76: E231 missing whitespace after ','
doc/source/categorical.rst:648:24: E231 missing whitespace after ','
doc/source/categorical.rst:655:34: E231 missing whitespace after ','
doc/source/categorical.rst:655:38: E231 missing whitespace after ','
doc/source/categorical.rst:655:42: E231 missing whitespace after ','
doc/source/categorical.rst:655:64: E231 missing whitespace after ','
doc/source/categorical.rst:655:68: E231 missing whitespace after ','
doc/source/categorical.rst:656:27: E231 missing whitespace after ':'
doc/source/categorical.rst:656:35: E231 missing whitespace after ','
doc/source/categorical.rst:656:39: E231 missing whitespace after ':'
doc/source/categorical.rst:656:44: E231 missing whitespace after ','
doc/source/categorical.rst:656:48: E231 missing whitespace after ','
doc/source/categorical.rst:656:52: E231 missing whitespace after ','
doc/source/categorical.rst:656:67: E231 missing whitespace after ':'
doc/source/categorical.rst:656:70: E231 missing whitespace after ','
doc/source/categorical.rst:656:72: E231 missing whitespace after ','
doc/source/categorical.rst:656:74: E231 missing whitespace after ','
doc/source/categorical.rst:674:24: E231 missing whitespace after ','
doc/source/categorical.rst:674:28: E231 missing whitespace after ','
doc/source/categorical.rst:674:32: E231 missing whitespace after ','
doc/source/categorical.rst:674:36: E231 missing whitespace after ','
doc/source/categorical.rst:674:40: E231 missing whitespace after ','
doc/source/categorical.rst:674:44: E231 missing whitespace after ','
doc/source/categorical.rst:674:48: E231 missing whitespace after ','
doc/source/categorical.rst:675:26: E231 missing whitespace after ','
doc/source/categorical.rst:675:30: E231 missing whitespace after ','
doc/source/categorical.rst:675:34: E231 missing whitespace after ','
doc/source/categorical.rst:675:38: E231 missing whitespace after ','
doc/source/categorical.rst:675:42: E231 missing whitespace after ','
doc/source/categorical.rst:675:46: E231 missing whitespace after ','
doc/source/categorical.rst:676:11: E225 missing whitespace around operator
doc/source/categorical.rst:676:15: E231 missing whitespace after ','
doc/source/categorical.rst:676:17: E231 missing whitespace after ','
doc/source/categorical.rst:676:19: E231 missing whitespace after ','
doc/source/categorical.rst:676:21: E231 missing whitespace after ','
doc/source/categorical.rst:676:23: E231 missing whitespace after ','
doc/source/categorical.rst:676:25: E231 missing whitespace after ','
doc/source/categorical.rst:677:30: E231 missing whitespace after ':'
doc/source/categorical.rst:677:35: E231 missing whitespace after ','
doc/source/categorical.rst:677:44: E231 missing whitespace after ':'
doc/source/categorical.rst:678:16: E231 missing whitespace after ','
doc/source/categorical.rst:679:16: E231 missing whitespace after ','
doc/source/categorical.rst:680:19: E231 missing whitespace after ','
doc/source/categorical.rst:696:13: E231 missing whitespace after ','
doc/source/categorical.rst:697:37: E231 missing whitespace after ','
doc/source/categorical.rst:697:41: E231 missing whitespace after ','
doc/source/categorical.rst:698:14: E231 missing whitespace after ','
doc/source/categorical.rst:698:22: E261 at least two spaces before inline comment
doc/source/categorical.rst:709:17: E231 missing whitespace after ','
doc/source/categorical.rst:763:24: E231 missing whitespace after ','
doc/source/categorical.rst:763:28: E231 missing whitespace after ','
doc/source/categorical.rst:763:32: E231 missing whitespace after ','
doc/source/categorical.rst:763:36: E231 missing whitespace after ','
doc/source/categorical.rst:763:40: E231 missing whitespace after ','
doc/source/categorical.rst:763:44: E231 missing whitespace after ','
doc/source/categorical.rst:764:31: E231 missing whitespace after ','
doc/source/categorical.rst:764:35: E231 missing whitespace after ','
doc/source/categorical.rst:764:39: E231 missing whitespace after ','
doc/source/categorical.rst:764:43: E231 missing whitespace after ','
doc/source/categorical.rst:764:47: E231 missing whitespace after ','
doc/source/categorical.rst:764:51: E231 missing whitespace after ','
doc/source/categorical.rst:764:73: E231 missing whitespace after ','
doc/source/categorical.rst:765:16: E231 missing whitespace after ','
doc/source/categorical.rst:765:18: E231 missing whitespace after ','
doc/source/categorical.rst:765:20: E231 missing whitespace after ','
doc/source/categorical.rst:765:22: E231 missing whitespace after ','
doc/source/categorical.rst:765:24: E231 missing whitespace after ','
doc/source/categorical.rst:765:26: E231 missing whitespace after ','
doc/source/categorical.rst:766:30: E231 missing whitespace after ':'
doc/source/categorical.rst:766:35: E231 missing whitespace after ','
doc/source/categorical.rst:766:44: E231 missing whitespace after ':'
doc/source/categorical.rst:768:16: E231 missing whitespace after ','
doc/source/categorical.rst:768:27: E231 missing whitespace after ','
doc/source/categorical.rst:768:30: E231 missing whitespace after ','
doc/source/categorical.rst:768:35: E231 missing whitespace after ','
doc/source/categorical.rst:771:20: E231 missing whitespace after ','
doc/source/categorical.rst:771:31: E231 missing whitespace after ','
doc/source/categorical.rst:771:34: E231 missing whitespace after ','
doc/source/categorical.rst:771:39: E231 missing whitespace after ','
doc/source/categorical.rst:779:19: E231 missing whitespace after ','
doc/source/categorical.rst:779:49: E231 missing whitespace after ','
doc/source/categorical.rst:779:71: E231 missing whitespace after ','
doc/source/categorical.rst:782:23: E231 missing whitespace after ','
doc/source/categorical.rst:782:53: E231 missing whitespace after ','
doc/source/categorical.rst:782:75: E231 missing whitespace after ','
doc/source/categorical.rst:782:79: E231 missing whitespace after ','
doc/source/categorical.rst:782:84: E501 line too long (80 > 79 characters)
doc/source/categorical.rst:790:27: E231 missing whitespace after ':'
doc/source/categorical.rst:790:30: E231 missing whitespace after ','
doc/source/categorical.rst:790:32: E231 missing whitespace after ','
doc/source/categorical.rst:790:34: E231 missing whitespace after ','
doc/source/categorical.rst:790:36: E231 missing whitespace after ','
doc/source/categorical.rst:790:44: E231 missing whitespace after ':'
doc/source/categorical.rst:790:49: E231 missing whitespace after ','
doc/source/categorical.rst:790:53: E231 missing whitespace after ','
doc/source/categorical.rst:790:57: E231 missing whitespace after ','
doc/source/categorical.rst:790:61: E231 missing whitespace after ','
doc/source/categorical.rst:791:15: E231 missing whitespace after ','
doc/source/categorical.rst:791:42: E231 missing whitespace after ','
doc/source/categorical.rst:791:64: E231 missing whitespace after ','
doc/source/categorical.rst:792:15: E231 missing whitespace after ','
doc/source/categorical.rst:792:42: E231 missing whitespace after ','
doc/source/categorical.rst:792:64: E231 missing whitespace after ','
doc/source/categorical.rst:806:25: E231 missing whitespace after ','
doc/source/categorical.rst:807:14: E231 missing whitespace after ','
doc/source/categorical.rst:808:30: E231 missing whitespace after ':'
doc/source/categorical.rst:808:42: E231 missing whitespace after ':'
doc/source/categorical.rst:809:24: E231 missing whitespace after ','
doc/source/categorical.rst:818:47: E231 missing whitespace after ','
doc/source/categorical.rst:820:22: E231 missing whitespace after ','
doc/source/categorical.rst:843:5: E402 module level import not at top of file
doc/source/categorical.rst:979:5: E402 module level import not at top of file
doc/source/categorical.rst:988:30: E231 missing whitespace after ':'
doc/source/categorical.rst:988:40: E231 missing whitespace after ':'
doc/source/categorical.rst:988:43: E231 missing whitespace after ','
doc/source/categorical.rst:988:45: E231 missing whitespace after ','
doc/source/categorical.rst:988:47: E231 missing whitespace after ','
doc/source/categorical.rst:988:49: E231 missing whitespace after ','
doc/source/categorical.rst:988:51: E231 missing whitespace after ','
doc/source/categorical.rst:996:84: E501 line too long (82 > 79 characters)
doc/source/categorical.rst:1065:24: E231 missing whitespace after ','
doc/source/categorical.rst:1065:31: E226 missing whitespace around arithmetic operator
doc/source/categorical.rst:1108:10: E111 indentation is not a multiple of four
doc/source/categorical.rst:1129:36: E231 missing whitespace after ','
doc/source/categorical.rst:1129:38: E231 missing whitespace after ','
doc/source/categorical.rst:1129:40: E231 missing whitespace after ','
doc/source/categorical.rst:1132:9: E265 block comment should start with '# '
doc/source/categorical.rst:1134:10: E111 indentation is not a multiple of four
doc/source/categorical.rst:1148:27: E231 missing whitespace after ':'
doc/source/categorical.rst:1148:30: E231 missing whitespace after ','
doc/source/categorical.rst:1148:32: E231 missing whitespace after ','
doc/source/categorical.rst:1148:34: E231 missing whitespace after ','
doc/source/categorical.rst:1149:27: E231 missing whitespace after ':'
doc/source/categorical.rst:1149:32: E231 missing whitespace after ','
doc/source/categorical.rst:1149:36: E231 missing whitespace after ','
doc/source/categorical.rst:1149:40: E231 missing whitespace after ','
doc/source/categorical.rst:1150:30: E231 missing whitespace after ':'
doc/source/categorical.rst:1150:48: E231 missing whitespace after ','
doc/source/categorical.rst:1150:50: E231 missing whitespace after ','
doc/source/categorical.rst:1150:52: E231 missing whitespace after ','
doc/source/categorical.rst:1167:29: E231 missing whitespace after ','
doc/source/categorical.rst:1167:31: E231 missing whitespace after ','
doc/source/categorical.rst:1167:33: E231 missing whitespace after ','
doc/source/categorical.rst:1167:51: E231 missing whitespace after ','
doc/source/categorical.rst:1167:53: E231 missing whitespace after ','
doc/source/categorical.rst:1167:55: E231 missing whitespace after ','
doc/source/categorical.rst:1168:19: E231 missing whitespace after ','
doc/source/categorical.rst:1168:23: E231 missing whitespace after ','
doc/source/categorical.rst:1168:27: E231 missing whitespace after ','
doc/source/categorical.rst:1169:16: E231 missing whitespace after ','
doc/source/categorical.rst:1169:18: E231 missing whitespace after ','
doc/source/categorical.rst:1169:20: E231 missing whitespace after ','
doc/source/categorical.rst:1170:33: E231 missing whitespace after ':'
doc/source/categorical.rst:1170:51: E231 missing whitespace after ':'
doc/source/categorical.rst:1184:28: E231 missing whitespace after ','
doc/source/categorical.rst:1184:30: E231 missing whitespace after ','
doc/source/categorical.rst:1184:32: E231 missing whitespace after ','
doc/source/categorical.rst:1184:51: E231 missing whitespace after ','
doc/source/categorical.rst:1184:53: E231 missing whitespace after ','
doc/source/categorical.rst:1184:55: E231 missing whitespace after ','
doc/source/categorical.rst:1184:57: E231 missing whitespace after ','
doc/source/categorical.rst:1190:34: E231 missing whitespace after ','
doc/source/categorical.rst:1190:36: E231 missing whitespace after ','
doc/source/categorical.rst:1190:38: E231 missing whitespace after ','
doc/source/categorical.rst:1190:40: E231 missing whitespace after ','
doc/source/categorical.rst:1197:28: E231 missing whitespace after ','
doc/source/categorical.rst:1197:30: E231 missing whitespace after ','
doc/source/categorical.rst:1197:32: E231 missing whitespace after ','
doc/source/categorical.rst:1197:51: E231 missing whitespace after ','
doc/source/categorical.rst:1197:53: E231 missing whitespace after ','
doc/source/categorical.rst:1197:55: E231 missing whitespace after ','
doc/source/categorical.rst:1197:57: E231 missing whitespace after ','
```

After fixing this issue, the previous command should not report any error.

### PR DESCRIPTION
**Pull request by charlesdong1991**
- [x] closes #23793
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: doc/source/categorical.rst [hunk 1]
```diff
@@ -7,6 +7,9 @@
 
    import numpy as np
    import pandas as pd
+   from pandas.api.types import CategoricalDtype
```
[the remark this answers is not in the document]
Comment (datapythonista) [reply doc/source/categorical.rst]: Agree, I think `flake8-rst` is reporting as errors having these imports in the blocks, but that should be ignored in the config: https://github.com/pandas-dev/pandas/pull/23802#issuecomment-440319546

[the remark this answers is not in the document]
Comment (charlesdong1991) [reply doc/source/categorical.rst]: even if moving back to where they were will cause an error message in flake8 check?

File: doc/source/categorical.rst [hunk 2]
```diff
@@ -1105,7 +1117,7 @@ NumPy itself doesn't know about the new `dtype`:
     try:
         np.dtype(dtype)
     except TypeError as e:
-         print("TypeError: " + str(e))
+        print("TypeError: " + str(e))
```
Comment (datapythonista) [inline doc/source/categorical.rst]:
```suggestion
        print("TypeError:", str(e))
```

File: doc/source/categorical.rst [hunk 3]
```diff
@@ -1126,12 +1138,12 @@ are not numeric data (even in the case that ``.categories`` is numeric).
...
 
-    s = pd.Series(pd.Categorical([1,2,3,4]))
+    s = pd.Series(pd.Categorical([1, 2, 3, 4]))
     try:
         np.sum(s)
-        #same with np.log(s),..
+        # same with np.log(s),..
     except TypeError as e:
-         print("TypeError: " + str(e))
+        print("TypeError: " + str(e))
```
Comment (datapythonista) [inline doc/source/categorical.rst]:
```suggestion
        print("TypeError:", str(e))
```

File: doc/source/categorical.rst [hunk 4]
```diff
@@ -1126,12 +1138,12 @@ are not numeric data (even in the case that ``.categories`` is numeric).
 
 .. ipython:: python
 
-    s = pd.Series(pd.Categorical([1,2,3,4]))
+    s = pd.Series(pd.Categorical([1, 2, 3, 4]))
     try:
         np.sum(s)
-        #same with np.log(s),..
+        # same with np.log(s),..
```
Comment (datapythonista) [inline doc/source/categorical.rst]:
```suggestion
        # same with np.log(s),...
```

**Review by datapythonista — COMMENTED** [review]
Looks great, added couple of small details as suggestions, and if you can restore the position of the imports, I think we're ready to merge.

File: doc/source/categorical.rst (hunk 3 above)
Comment (charlesdong1991) [reply doc/source/categorical.rst]: thanks, changed this and applied the same to others.

File: doc/source/categorical.rst (hunk 4 above)
Comment (charlesdong1991) [reply doc/source/categorical.rst]: changed

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @charlesdong1991

File: doc/source/categorical.rst [hunk 5]
```diff
@@ -671,13 +681,14 @@ If the slicing operation returns either a ``DataFrame`` or a column of type
...
-    df.iloc[2:4,:].dtypes
-    df.loc["h":"j","cats"]
+    idx = pd.Index(["h", "i", "j", "k", "l", "m", "n"])
+    cats = pd.Series(["a", "b", "b", "b", "c", "c", "c"],
+                     dtype="category", index=idx)
+    values = [1, 2, 2, 2, 3, 4, 5]
+    df = pd.DataFrame({"cats": cats, "values": values}, index=idx)
+    df.iloc[2:4, :]
+    df.iloc[2:4, :].dtypes
+    df.loc["h": "j", "cats"]
```
Comment (WillAyd) [inline doc/source/categorical.rst]: Hmm this doesn’t seem right - no need for spacing after colon, no?

File: doc/source/categorical.rst [hunk 6]
```diff
@@ -760,36 +771,38 @@ value is included in the `categories`:
...
     except ValueError as e:
-        print("ValueError: " + str(e))
+        print("ValueError:", str(e))
 
 Setting values by assigning categorical data will also check that the `categories` match:
 
 .. ipython:: python
 
-    df.loc["j":"k","cats"] = pd.Categorical(["a","a"], categories=["a","b"])
+    df.loc["j": "k", "cats"] = pd.Categorical(["a", "a"], categories=["a", "b"])
```
Comment (WillAyd) [inline doc/source/categorical.rst]: Same comment here

File: doc/source/categorical.rst (hunk 5 above)
Comment (charlesdong1991) [reply doc/source/categorical.rst]: ahh, yeah, right!! thanks for the check!! sorry for my carelessness...

**Review by WillAyd — APPROVED** [review]
Ping on green

**Comment by datapythonista** [comment]
Thanks a lot for these fixes @charlesdong1991

**Pull request merged**